Fix: Consistent sidebar width across tabs in Publisher Portal (#3617)

ProgrammingPirates opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/wso2/apim-apps?pullRequest=1155) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/wso2/apim-apps?pullRequest=1155) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/wso2/apim-apps?pullRequest=1155) it.</sub>

ProgrammingPirates commented:
> Hi @ProgrammingPirates,
> 
> I have applied your latest changes locally and checked the behaviour. I still see the issue I reported in comment [1]. Shall we fix that?
> 
> [1] [#1155 (comment)](https://github.com/wso2/apim-apps/pull/1155#issuecomment-3425031543)
> 
> Thanks, Tharika.

ok maam wait

ProgrammingPirates commented:
@tharikaGitHub check it

tharikaGitHub commented:
Hi @ProgrammingPirates,

Seems all changes are reverted back to the state in [1]. I can again see the scrolling issue.

[1] https://github.com/wso2/apim-apps/pull/1155#issuecomment-3425031543

Thanks,
Tharika.

ProgrammingPirates commented:
@tharikaGitHub pls maam be active here

ProgrammingPirates commented:
> Hi @ProgrammingPirates,
> 
> I appreciate your enthusiasm. Thank you for making an active effort to contribute to our code base. Instead of both of us having to go back and forth on this issue, it would be really nice if you can test locally the following, before making the next commit.
> 
> 1. Check if the original issue is fixed. (Side bar width varying between tabs)
> 2. Check if the UI does not cut off from the right end of the browser window.
> 
> **As you have suggested, I tested the latest changes locally on my machine. However, unfortunately both the above points 1 and 2 are not fixed.**
> 
> The following is my suggestion.
> 
> 1. The changes you have done to the `Comments.jsx` and `defaultTheme.js` files are unnecessary. So you can remove them.
> 2. The changes you have done to `index.jsx` file should be modified so that the UI does not cut off from the right end of the browser window.
> 
> Shall we make the necessary changes and test your fix locally before making the next commit? Then I can review and approve your PR.
> 
> Thanks, Tharika.

i try to run localhost but its take too much time to loading ans server not repond
<img width="1874" height="589" alt="image" src="https://github.com/user-attachments/assets/bbb21648-1831-4cc5-85bf-ce7e0cfe3fb2" />

ProgrammingPirates commented:
**Hi @tharikaGitHub,

Thank you for the detailed feedback! I've implemented the fix based on your suggestions:

Changes Made:

defaultTheme.js: Fixed drawerWidth: 230 to match leftMenuWidth: 230 for consistent sidebar width
index.jsx: Enhanced content width calculation to prevent UI cutoff:
Uses min(calc(100vw - leftMenuWidth), contentAreaWidth)
Added boxSizing: border-box for proper padding calculation
This should fix both issues:

✅ Sidebar width now consistent across all tabs
✅ Content no longer cuts off from right end of browser window
The changes are ready for your local testing. Please test and let me know if this resolves the issues!

Thanks for your patience and guidance**

tharikaGitHub commented:
Hi @ProgrammingPirates,

Now your fix works correctly. Thank you for your contribution! Let's do the following as the final touches to your PR.

1. Remove changes done to the following files which are not necessary for this fix.

```
- EndpointListing.jsx
- EndpointOverview.jsx
- GenericEndpoint.jsx
- LoadbalanceFailoverConfig.jsx
- defaultTheme.js
```

2. Remove the following line from the `index.jsx` file as it adds an unnecessary padding on the right side of the window. 

`boxSizing: 'border-box', // Include padding in width calculation`

With the above line under the css `[`& .${classes.contentInside}`]: {`, the UI adds a padding on the right side as you can see below.

<img width="1511" height="768" alt="Screenshot 2025-10-25 at 13 11 11" src="https://github.com/user-attachments/assets/3d2d0a5b-8dd0-48c8-824f-f1b21e59fbf2" />

Without the `boxSizing` line, it looks as follows which is more appropriate. So let's remove it.

<img width="1511" height="765" alt="Screenshot 2025-10-25 at 13 10 19" src="https://github.com/user-attachments/assets/aa277967-7a79-4b2b-8242-cd8a28d5c6db" />


**So in the end your PR should only contain changes to the `index.jsx` after removing `boxSizing: 'border-box', // Include padding in width calculation` line from the css styling.**

Shall we make these final changes?

Thanks,
Tharika.

ProgrammingPirates commented:
> Hi @ProgrammingPirates,
> 
> Thanks, Shall we remove the changes to the following files as well as mentioned in comment [1]? I can still see them in the PR.
> 
> ```
> - EndpointListing.jsx
> - EndpointOverview.jsx
> - GenericEndpoint.jsx
> - LoadbalanceFailoverConfig.jsx
> - OperationGovernance.jsx
> ```
> 
> [1] [#1155 (comment)](https://github.com/wso2/apim-apps/pull/1155#issuecomment-3446130366)
> 
> Thanks, Tharika.

check

tharikaGitHub commented:
Great! now you only have the required changes. I'll approve the PR. Thank you for your contribution @ProgrammingPirates!

tharikaGitHub commented:
Hi @ProgrammingPirates,

I have noticed today that with this fix, the UI does not render properly in larger screens. There is a large padding added to the right side of the screen as shown below. This was not the case without the fix.

<img width="1920" height="947" alt="Screenshot 2025-10-26 at 15 43 35" src="https://github.com/user-attachments/assets/31c66a8e-9fdd-4320-9687-13e0aa43585f" />

Shall we correct the fix considering the responsiveness aspect as well?

Thanks,
Tharika.

ProgrammingPirates commented:
> @ProgrammingPirates, seems unnecessary commits have been merged again. Shall we clear them out and fix the issue? There are 13 files in the PR now.

check pls

tharikaGitHub commented:
Hi @ProgrammingPirates,

This PR contains 10 other files which are not necessary for the fix of this issue. We only need the change done to [1] including the fix for the concern raised in comment [2]. Will you be able to kindly fix the issue and submit only the relevant file in this PR?

Please follow the steps below if you find it difficult to commit only the relevant change. 

1. Fetch the latest changes from the upstream of apim-apps repo.
```
git fetch upstream main
```
2. Checkout a new branch `side-bar-width-issue` from main
```
git checkout -b side-bar-width-issue upstream/main
```
3. Apply only the changes in file [1] including the fix for the concern raised in [2] as well.
4. Add the changes done to `index.jsx`. Replace the \<path to index.jsx\> in the following command with the correct path to the file.
```
git add <path to index.jsx>
```
5. Commit the changes.
```
git commit -m "Fix issue with inconsistent side bar width"
```
6. Push the changes to the branch `side-bar-width-issue`. Make sure the origin is your forked gitHub repo link.
```
git push origin side-bar-width-issue
```
7. Open a new PR from the `side-bar-width-issue` branch in your gitHub repo of apim-apps.

[1] https://github.com/wso2/apim-apps/pull/1155/files#diff-13c3ddccfaeec727419ee91b77fdefb21fe2f546c482f60083699516bce46925R129-R136
[2] https://github.com/wso2/apim-apps/pull/1155#issuecomment-3448346449

Thanks,
Tharika.

ProgrammingPirates commented:
> Hi @ProgrammingPirates,
> 
> This PR contains 10 other files which are not necessary for the fix of this issue. We only need the change done to [1] including the fix for the concern raised in comment [2]. Will you be able to kindly fix the issue and submit only the relevant file in this PR?
> 
> Please follow the steps below if you find it difficult to commit only the relevant change.
> 
> 1. Fetch the latest changes from the upstream of apim-apps repo.
> 
> ```
> git fetch upstream main
> ```
> 
> 2. Checkout a new branch `side-bar-width-issue` from main
> 
> ```
> git checkout -b side-bar-width-issue upstream/main
> ```
> 
> 3. Apply only the changes in file [1] including the fix for the concern raised in [2] as well.
> 4. Add the changes done to `index.jsx`. Replace the <path to index.jsx> in the following command with the correct path to the file.
> 
> ```
> git add <path to index.jsx>
> ```
> 
> 5. Commit the changes.
> 
> ```
> git commit -m "Fix issue with inconsistent side bar width"
> ```
> 
> 6. Push the changes to the branch `side-bar-width-issue`. Make sure the origin is your forked gitHub repo link.
> 
> ```
> git push origin side-bar-width-issue
> ```
> 
> 7. Open a new PR from the `side-bar-width-issue` branch in your gitHub repo of apim-apps.
> 
> [1] https://github.com/wso2/apim-apps/pull/1155/files#diff-13c3ddccfaeec727419ee91b77fdefb21fe2f546c482f60083699516bce46925R129-R136 [2] [#1155 (comment)](https://github.com/wso2/apim-apps/pull/1155#issuecomment-3448346449)
> 
> Thanks, Tharika.

can u tell me file name, which i need to remove ?

tharikaGitHub commented:
Hi @ProgrammingPirates,

You can create a different branch for each issue.

For fixing this issue, you can follow the steps mentioned in [1] to create a new branch and proceed. Then when you want to open a PR, open it using the relevant branch.

You can follow the same for the other issue in a **different branch**.

Let me know if you need any further clarification.

[1] https://github.com/wso2/apim-apps/pull/1155#issuecomment-3471339517

ProgrammingPirates commented:
> > Ma’am, if I’m changing code locally and need the same change in two places, what should I do?
> 
> 1. You can first create 2 different branches using the upstream/main. (Assume they are called `branch-a` and `branch-b`)
> 2. Then do the change you need on one of the branches. Say `branch-a`.
> 3. Then commit the change to `branch-a` like you normally do.
> 4. Now you can push the change to the `branch-a` branch of your origin repo.
> 5. After that, checkout to `branch-b` locally using `git checkout branch-b` command.
> 6. You can use the cherry-pick command to apply the same fix you did to `branch-a` on top of `branch-b` now.
> 7. Then you can push the changes to the `branch-b` branch of your origin repo.
> 
> You can refer to article [1] to refer to any command you need.
> 
> [1] https://medium.com/@tharika/the-git-cheat-sheet-for-most-common-operations-ecf720ffcf0c

ok maam thanks

tharikaGitHub commented:
Hi @ProgrammingPirates,

I checked your fix by applying it locally. However the initial issue of the sidebar and also the cutting off of the UI from the right end has not been fixed. Shall we check it again?

Thanks,
Tharika.

tharikaGitHub commented:
> ok can u send ss ?

Sure.

<img width="1922" height="862" alt="Screenshot 2025-10-31 at 13 21 00" src="https://github.com/user-attachments/assets/1cf96d1e-d445-4b43-83b6-a66599364958" />


<img width="1919" height="830" alt="Screenshot 2025-10-31 at 13 22 19" src="https://github.com/user-attachments/assets/7b743d14-b0c9-4672-8278-e0c7e1083a57" />

ProgrammingPirates commented:
> Hi @ProgrammingPirates,
> 
> I applied it now. But still observe the same issue.

 [`& .${classes.content}`]: {
        display: 'flex',
        flexGrow: 1,
        flexDirection: 'column',
        paddingBottom: theme.spacing(3),
        overflow: 'auto',
        minWidth: 0, // Allow flex child to shrink below content size
    },

    [`& .${classes.contentInside}`]: {
        width: '100%',
        maxWidth: '100%', // Constrain to parent width
        paddingLeft: theme.spacing(3),
        paddingRight: theme.spacing(3),
        paddingTop: theme.spacing(2),
    },
    can u check this

ProgrammingPirates commented:
<img width="1919" height="626" alt="image" src="https://github.com/user-attachments/assets/f6f9bf47-8092-4438-98f0-aad045fea072" />
how much time take to start fronted?

tharikaGitHub commented:
Hi @ProgrammingPirates 

> how much time take to start fronted?

Seems you are still trying to start using `npm start` command. Please use the instructions in [1] which would not take much time as this.

[1] https://medium.com/@tharika/how-to-apply-changes-to-wso2-api-manager-react-portals-on-the-go-d1cbb50f3819

ProgrammingPirates commented:
overflowX: 'hidden', // prevent horizontal scrolling
    minWidth: 0, // allow flex child to shrink below contetn size
    maxWidth: `min(calc(100vw - ${theme.custom.leftMenuWidth}px), ${theme.custom.contentAreaWidth}px)`, 
    // constain to viewport minus sidebar, respect contentAreaWidth
},

can u try once pls
